packages/@aws-cdk/aws-iotevents/README.md

```diff
@@ -53,52 +52,168 @@ const input = new iotevents.Input(this, 'MyInput', {
   inputName: 'my_input', // optional
   attributeJsonPaths: ['payload.deviceId', 'payload.temperature'],
 });
+```
 
-const warmState = new iotevents.State({
-  stateName: 'warm',
+To grant permissions to put messages in the Input,
+you can use the `grantWrite()` method:
+
+```ts
+import * as iam from '@aws-cdk/aws-iam';
+import * as iotevents from '@aws-cdk/aws-iotevents';
+
+declare const grantable: iam.IGrantable;
+declare const input: iotevents.IInput;
+
+input.grantWrite(grantable);
+```
+
+## State
+
+You can create `State` as following.
+If a State is used for a Detector Model's initial State, it's required that its `onEnter` Event is non-null,
+and contains a reference to an Input via the `condition` property.
+And if a message is put to the Input, the detector instances are created regardless of the evaluation result of `condition`.
+You can set the reference to Input with `iotevents.Expression.currentInput()` or `iotevents.Expression.inputAttribute()`.
+In other states, `onEnter` is optional.
+
+```ts
+import * as iotevents from '@aws-cdk/aws-iotevents';
+
+declare const input: iotevents.IInput;
+
+const initialState = new iotevents.State({
+  stateName: 'MyState',
   onEnter: [{
-    eventName: 'test-event',
+    eventName: 'onEnter',
     condition: iotevents.Expression.currentInput(input),
   }],
 });
-const coldState = new iotevents.State({
-  stateName: 'cold',
+```
+
+You can set actions on the `onEnter` event. They are performed if `condition` evaluates to `true`.
+If you omit `condition`, actions are performed every time the State is entered.
+For more information, see [supported actions](https://docs.aws.amazon.com/iotevents/latest/developerguide/iotevents-supported-actions.html).
+
+```ts
+import * as iotevents from '@aws-cdk/aws-iotevents';
+
+declare const input: iotevents.IInput;
+
+const setTemperatureAction = {
+  renderActionConfig: () => ({
+    configuration: {
+      setVariable: { 
+        variableName: 'temperature', 
+        value: iotevents.Expression.inputAttribute(input, 'payload.temperature').evaluate(),
+      },
+    },
+  }),
+};
```

Comment:
Let's not use these "inline" Actions. Let's show a real class from the `@aws-cdk/aws-iotevents-actions` package instead.